### ISSUE TITLE
An email address may not be changed once a customer is created

### ISSUE THREAD

**Issue by SantiParagarino** [mailchimp/mc-magento#17]

**Comment by twoonesixdigital** [comment]
Hi guys. I was curious how things are progressing with this issue. It just became extremely relevant to us with our Miva/MailChimp integration and we're kind of on hold pending its resolution. Can we expect that the API is going to be modified so that PUT requests to the customer resource containing a different email address will result in the customer record being updated rather than the error response we're currently getting? Thank you kindly!

**Comment by takzilla** [comment]
Hey there,

MailChimp dev here. We haven't been able to build this out yet on the MailChimp-side, so this issue is still unresolved and in progress. Apologies for the wait - I'll be sure to post an update once it's become a feature of our API.

**Comment by takzilla** [comment]
We're going to re-visit this in the coming weeks and I'll post our progress. Going to bump this up in priority as it's beginning to cause some headaches for other platform integrations as well.

**Comment by WinstonN** [comment]
Bump!

**Comment by brendanronan** [comment]
Bump! Any updates @takzilla ?

**Comment by szewczykmira** [comment]
Bump! @takzilla any updates?

**Comment by takzilla** [comment]
Sorry for the late response! No updates unfortunately - we don't have any concrete plans in the short-term to complete work on this. If that changes I'll be sure to update this thread.

**Comment by danharper83** [comment]
@Santiagoebizmarts what is the workaround if a customer has changed their email address? I have lots of errors but can't work out how to reset or remove the user to prevent the errors. The error log doesn't say which user it is.

**Comment by Santiagoebizmarts** [comment]
@danharper83 
For customers who are subscribed we remove the subscriber and create a new one.
Unfortunately for the customer object used in Automations this is not done because you would loose all the history you have on that customer. Unfortunately there is not much we can do with this.

Regards.

**Comment by danharper83** [comment]
@Santiagoebizmarts the error log doesn't tell me which user it is that is failing.

**Comment by Santiagoebizmarts** [comment]
Hello,

This should only affect those particular cases but the rest should work normally. I am not sure what you mean with your last comment.

Regards.

**Comment by WinstonN** [comment]
This commit has been tested and works & will fix this. 
If the user changes his/her email address on the magento (on front or by admin) it will unsubscribe the original email and subscribe the new email. 
https://github.com/mailchimp/mc-magento/pull/520

**Comment by seansan** [comment]
ALSO:
mailchimp just replied and said email can be updated right here (and this is the advised way)

https://developer.mailchimp.com/documentation/mailchimp/reference/lists/members/#edit-patch_lists_list_id_members_subscriber_hash

Use PATCH command and change email

**Comment by Santiagoebizmarts** [comment]
@seansan 

You are right about subscribers emails, unfortunately for [customers](https://developer.mailchimp.com/documentation/mailchimp/reference/ecommerce/stores/customers/#edit-patch_ecommerce_stores_store_id_customers_customer_id) this is not possible yet.

Regards.

**Comment by seansan** [comment]
got it!

and it is not possible to use a LIST command here? (maybe even as a workaround)

**Comment by Santiagoebizmarts** [comment]
@seansan Unfortunately it is not. I just tested if the change in the subscriber endpoint affected the customer and it doesn't.

**Comment by seansan** [comment]
@Santiagoebizmarts 

MC keeps confirming that is IS possible. When I said it is impossible they repsonded with this answer

"Make a GET call to the /lists/{list_id}/members endpoint.  After making that call, you can get the subscriber_hash value for the list member you want to update and use it with the PATCH call to change the target subscribers email address.  "

**Comment by brendanronan** [comment]
@seansan the specific issue we're referring to applies to [customers](https://developer.mailchimp.com/documentation/mailchimp/reference/ecommerce/stores/customers/), not [list members](https://developer.mailchimp.com/documentation/mailchimp/reference/lists/members/). They are two different entities.

**Comment by tompicca** [comment]
At today, once a customer is created with a given email, there's no way to change its email. Also, is not possible to delete the customer and create a new one with an update email because customer may only be deleted if no orders are associated with that customer. This is a serious lack for our ecommerce. Someone knows a workaround for this? @takzilla , what is the official procedure according to MC?

**Comment by pelargir** [comment]
+1, we are having the same problem. Customers can and do change their email addresses in our system, and that change needs to be reflected in Mailchimp. Sending a PATCH to update the list member that's associated with the customer results in this error:

> This member's status is "transactional." You can only update email addresses for members with a status of "subscribed."

So we're stuck at the moment. No way to update a customer's email address without first deleting all of their orders, then deleting the customer and re-creating ALL of their orders.
